lib/public/Server.php

```diff
@@ -24,12 +25,9 @@
  */
 final class Server {
 	/**
-	 * @template T
-	 * @param class-string<T>|string $serviceName
-	 * @return T|mixed
-	 * @psalm-template S as class-string<T>|string
-	 * @psalm-param S $serviceName
-	 * @psalm-return (S is class-string<T> ? T : mixed)
+	 * @psalm-template T
+	 * @psalm-param class-string<T>|string $serviceName
+	 * @psalm-return ($serviceName is class-string<T> ? T : mixed)
```

Comment:
```suggestion
	 * @psalm-param class-string<T> $serviceName
	 * @psalm-return T
```
Isn't this enough? If the class isn't found then an exception is thrown, so the return should be irrelevant.

Reply:
But it can also be an alias instead of just a class name, right?

Reply:
Does not really work. I think this version is already simpler and works :)